### PR TITLE
app: check sim status on startup

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch checks whether a SIM card is inserted on startup. An error is printed if it's missing.
This is to guide the user to avoid some troubleshooting.

### CODE REVIEW

**Comment by simensrostad** [comment]
Could also check NW_REG_STATUS event in link controller handler.

```c
case LTE_LC_EVT_NW_REG_STATUS: {
		if (evt->nw_reg_status == LTE_LC_NW_REG_UICC_FAIL) {
			LOG_ERR("No SIM card detected!");
			SEND_ERROR(modem, MODEM_EVT_ERROR, -ENOTSUP);
			break;
		}
```

**Comment by simensrostad** [comment]
> Could also check NW_REG_STATUS event in link controller handler.
> 
> ```c
> case LTE_LC_EVT_NW_REG_STATUS: {
> 		if (evt->nw_reg_status == LTE_LC_NW_REG_UICC_FAIL) {
> 			LOG_ERR("No SIM card detected!");
> 			SEND_ERROR(modem, MODEM_EVT_ERROR, -ENOTSUP);
> 			break;
> 		}
> ```

Should be pretty straight forward, just call lte_lc_register_handler, pass in the handler and check the event.

File: README.md [hunk 1]
```diff
@@ -44,7 +44,13 @@ west config build.sysbuild True
 To build the application, run the following command:
 
 ```shell
-west build -b thingy91x/nrf9151/ns app
+west build -b thingy91x/nrf9151/ns --sysbuild app
```
Comment (simensrostad) [inline README.md:47]: Is this step nessecary when we instruct the user to call:

```
west config build.sysbuild True
```

File: app/src/common/message_channel.h [hunk 2]
```diff
@@ -80,6 +87,7 @@ enum time_status {
 
 enum error_type {
 	ERROR_FATAL = 0x1,
+	ERROR_IRRECOVERABLE = 0x2,
```
Comment (simensrostad) [inline app/src/common/message_channel.h:90]:
```suggestion
	ERROR_IRRECOVERABLE,
```

File: app/src/modules/network/network.c [hunk 3]
```diff
@@ -123,6 +123,85 @@ static void connectivity_event_handler(struct net_mgmt_event_callback *cb,
 	}
 }
 
+static void lte_lc_evt_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS: {
```
Comment (simensrostad) [inline app/src/modules/network/network.c:129]: We only need this case, the other ones can be ignored. Future improvement would be to add this check to the nrf_modem_net_if implementation.

File: README.md (hunk 1 above)
Comment (maxd-nordic) [reply README.md]: you're right, I removed it.

**Review by bjda — APPROVED** [review]
Makes sense to me

**Pull request merged**